packages/volto/src/components/manage/Widgets/ButtonsWidget.tsx

```diff
@@ -0,0 +1,166 @@
+import Icon from '@plone/volto/components/theme/Icon/Icon';
+import { Radio, RadioGroup } from '@plone/components';
+import isEqual from 'lodash/isEqual';
+import type { StyleDefinition } from '@plone/types';
+
+/**
+ * A tuple that has an icon in the first element and a i18n string in the second.
+ */
+export type ActionInfo = [React.ReactElement<any>, string] | [string, string];
+
+type ActionValue = string | Record<`--${string}`, string>;
+
+export type ButtonsWidgetProps = {
+  /**
+   * Unique identifier for the widget.
+   */
+  id: string;
+
+  /**
+   * Callback function to handle changes.
+   */
+  onChange: (id: string, value: ActionValue) => void;
+
+  /**
+   * List of actions available for the widget.
+   */
+  actions?: Array<StyleDefinition | string>;
+
+  /**
+   * Map containing additional the information (icon and i18n string) for each action.
+   */
+  actionsInfoMap?: Record<string, ActionInfo>;
+
+  /**
+   * List of actions to be filtered out. In case that we don't want the default ones
+   * we can filter them out.
+   */
+  filterActions?: string[];
+
+  /**
+   * Current value of the widget.
+   */
+  value?: ActionValue;
+
+  /**
+   * Default value of the widget.
+   */
+  default?: ActionValue;
+
+  /**
+   * Indicates if the widget is disabled.
+   */
+  disabled?: boolean;
+
+  /**
+   * Indicates if the widget is disabled (alternative flag for compatibility reasons).
+   */
+  isDisabled?: boolean;
+  [key: string]: any;
+};
+
+type NormalizedAction = {
+  name: string;
+  value: ActionValue;
+};
+
+const ButtonsWidget = (props: ButtonsWidgetProps) => {
+  const {
+    disabled,
+    id,
+    onChange,
+    actions = [],
+    actionsInfoMap,
+    value,
+    isDisabled,
+    default: defaultValue,
+  } = props;
+  const normalizedActions = React.useMemo<NormalizedAction[]>(
+    () =>
+      actions.map((action) =>
+        typeof action === 'string'
+          ? { name: action, value: action }
+          : {
+              name: action.name,
+              value: action.style ?? action.name,
+            },
+      ),
+    [actions],
+  );
+
+  const selectedActionName = React.useMemo(
+    () =>
+      normalizedActions.find((action) => isEqual(value, action.value))?.name,
+    [normalizedActions, value],
+  );
+
+  const handleChange = React.useCallback(
+    (selectedName: string) => {
+      const selectedAction = normalizedActions.find(
+        ({ name }) => name === selectedName,
+      );
+
+      if (selectedAction) {
+        onChange(id, selectedAction.value);
+      }
+    },
+    [id, normalizedActions, onChange],
+  );
+
+  React.useEffect(() => {
+    if (!value && defaultValue) {
+      const nextValue =
+        typeof defaultValue === 'string'
+          ? normalizedActions.find(({ name }) => name === defaultValue)
+              ?.value ?? defaultValue
+          : defaultValue;
+
+      onChange(id, nextValue);
+    }
+  }, [defaultValue, id, normalizedActions, onChange, value]);
```

Comment:
What is this for?

Reply:
It was necessary for managing the default value... This predates the "fixed" defaults in the blocks settings, so I haven't tried if that's necessary afterwards. Also, could be that RAC takes care for you, so you don't have to now. I will take a look.

Reply:
I removed it, and refactored to use RadioGroup `defaultValue` instead.